### DISCUSSION TITLE
原创文章-- Sui Object 的安全模型分析

### DISCUSSION THREAD

**Discussion by jolestar** [starcoinorg/web3startrek#118] [Call for Article]
Sui Move 引入 Object 后，对原 Move 类型安全方面做了一些改变，也会引入新的风险。

这篇文章分析它的安全模型，以及原理，提供一些开发者的注意事项，避免安全问题。

1. Move 本身的安全模型介绍（ struct 的可见性，ability, move_to|move_from|borrow_global 指令的类型约束）
2. Sui Object 的安全模型（对 Object 的 ability(key, store)要求，对 verifier 的扩展）
3. Sui Object 的 owner 模型（owner by address， owner by object, share, immutable)
4. Sui 对交易参数的扩展和安全约束（以 Object 作为参数的校验条件）
5. Sui 对 Object 的操作的安全约束（transfer, public_transfer 等）
6. Sui Object 编程安全要点

**Comment by 0xyilu** [comment]
我们认领下
